.github/workflows/ci.yaml

```
@@ -0,0 +1,32 @@
+name: Python Unit Tests with Coverage
+
+on:
+  push:
+    branches: [ unit_tests ]
+  pull_request:
+    branches: [ unit_tests ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest

```

```suggestion
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ["3.10", "3.11", "3.12", "3.13"]
```
Since this library could potentially be used by a number of python version, it might be worth testing all possible versions. Because the type union syntax (e.g. `str | None`) which is used extensively in this repo, was only added in 3.10, 3.10 is the minimum supported version.